examples/fuse_examples/imaging/classification/mnist/run_mnist_custom_pl_imp.py

```diff
@@ -0,0 +1,377 @@
+        self._model = ModelWrapSeqToDict(model=torch_model,
+                            model_inputs=["data.image"],
+                            post_forward_processing_function=perform_softmax,
+                            model_outputs=['logits.classification', 'output.classification']
+                            )
+
+        # losses
+        self._losses = {
+            'cls_loss': LossDefault(pred='model.logits.classification', target='data.label', callable=F.cross_entropy, weight=1.0),
+        }
+
+        # metrics
+        self._train_metrics = OrderedDict([
+            ('operation_point', MetricApplyThresholds(pred='model.output.classification')), # will apply argmax
+            ('accuracy', MetricAccuracy(pred='results:metrics.operation_point.cls_pred', target='data.label'))
+        ])
+
+        self._validation_metrics = copy.deepcopy(self._train_metrics) # use the same metrics in validation as well
+        
+        
+    
+    ## forward
+    def forward(self, batch_dict: NDict) -> NDict:
+        return self._model(batch_dict)
+    
+    ## Step
+    def training_step(self, batch_dict: NDict, batch_idx: int) -> dict:
+        # run forward function and store the outputs in batch_dict["model"]
+        batch_dict["model"] = self.forward(batch_dict)
+        # given the batch_dict and FuseMedML style losses - compute the losses, return the total loss and save losses values in batch_dict["losses"]
+        total_loss = step_losses(self._losses, batch_dict)
+        # given the batch_dict and FuseMedML style losses - collect the required values to compute the metrics on epoch_end
+        step_metrics(self._train_metrics, batch_dict)
+
+        # return the total_loss, the losses and drop everything else
+        return {"loss": total_loss, "losses": batch_dict["losses"]} 
+     
+    def validation_step(self, batch_dict: NDict, batch_idx: int) -> dict:
+        # run forward function and store the outputs in batch_dict["model"]
+        batch_dict["model"] = self.forward(batch_dict)
+        # given the batch_dict and FuseMedML style losses - compute the losses, return the total loss (ignored) and save losses values in batch_dict["losses"]
+        _ = step_losses(self._losses, batch_dict)
+        # given the batch_dict and FuseMedML style losses - collect the required values to compute the metrics on epoch_end
+        step_metrics(self._validation_metrics, batch_dict)
+
+        # return just the losses and drop everything else
+        return {"losses": batch_dict["losses"]} 
+     
+    def predict_step(self, batch_dict: NDict, batch_idx: int) -> dict:
+        # run forward function and store the outputs in batch_dict["model"]
+        batch_dict["model"] = self.forward(batch_dict)
+        # extract the requried keys - defined in self.set_predictions_keys()
+        return step_extract_predictions(self._prediction_keys, batch_dict)
```

Comment:
perhaps an error if self._prediction_keys attribute is not found?
with instruction to call set_predictions_keys()